plugin/src/main/java/org/opensearch/ml/processor/MLInferenceIngestProcessor.java

```diff
@@ -162,10 +186,44 @@ private void processPredictions(
         List<Map<String, String>> processOutputMap,
         int inputMapIndex,
         int inputMapSize
-    ) {
+    ) throws IOException {
         Map<String, String> modelParameters = new HashMap<>();
+        Map<String, String> modelConfigs = new HashMap<>();
+
         if (inferenceProcessorAttributes.getModelConfigMaps() != null) {
             modelParameters.putAll(inferenceProcessorAttributes.getModelConfigMaps());
+            modelConfigs.putAll(inferenceProcessorAttributes.getModelConfigMaps());
+        }
+        Map<String, String> outputMapping = processOutputMap.get(inputMapIndex);
+
+        Map<String, Object> ingestDocumentSourceAndMetaData = new HashMap<>();
+        ingestDocumentSourceAndMetaData.putAll(ingestDocument.getSourceAndMetadata());
+        ingestDocumentSourceAndMetaData.put(IngestDocument.INGEST_KEY, ingestDocument.getIngestMetadata());
+
+        Map<String, List<String>> newOutputMapping = new HashMap<>();
+        for (Map.Entry<String, String> entry : outputMapping.entrySet()) {
+            String newDocumentFieldName = entry.getKey();
+            List<String> dotPathsInArray = writeNewDotPathForNestedObject(ingestDocumentSourceAndMetaData, newDocumentFieldName);
+            newOutputMapping.put(newDocumentFieldName, dotPathsInArray);
+        }
+
+        for (Map.Entry<String, String> entry : outputMapping.entrySet()) {
+            String newDocumentFieldName = entry.getKey();
+            List<String> dotPaths = newOutputMapping.get(newDocumentFieldName);
+
+            int existingFields = 0;
+            for (String path : dotPaths) {
+                if (ingestDocument.hasField(path)) {
+                    existingFields++;
+                }
+            }
+            if (!override && existingFields == dotPaths.size()) {
```

Comment:
if override is false and existing Fields is the same as the dothPath size, we skip adding the output mapping in silent ?

Reply:
Yes, it means if the ingest document had a field that already has the processor out field, e.g. text_embedding field, we can skip it in current running processor. User can explicitly set override to true to rewrite the output field though

Reply:
how does the user know if the field is skipped? Maybe we can add some logging.

Reply:
sure will do that